coverage/docs/design.md

```diff
@@ -0,0 +1,113 @@
+#Overview
+This code coverage tool has two major features.
+1. As a pre-submit tool, it runs code coverage on every single commit to Github and reports coverage change back to the PR as a comment by a robot account. It also has the ability to block a PR from merging if coverage falls below threshold
+2. As a post-submit / periodical running job, it reports on TestGrid to show users how coverage changes over time.
+
+##Users
+The presubmit tool is intended for a developer to see the impact on code coverage of his/her commit. It can also be used by repo managers to block any PR from merging if the coverage falls under customized threshold.
```

Comment:
This wording is confusing, it makes it sounds like you have implemented multiple different tools when in reality you have a single job that is running in multiple places.

Also like Steve noted, the job doesn't do anything to block a PR from merging, that functionality is provided by Tide. I think what you really mean is that job will fail if coverage falls below the configured threshold.

Reply:
Done.